### PR TITLE
Zelcash Integration

### PR DESCRIPTION
**Pull request by Cabecinha84**
After the last PR deny because of the copy paste code, I have refactor/redone the code to use Zcash implementation instead of copy/past to Zelcash classes.

### CODE REVIEW

**Comment by hewigovens** [comment]
can you rebase and resolve conflicts?

**Comment by Cabecinha84** [comment]
Sure.
When I have some free time this week I will do it and keep you posted.
Thanks.
Best regards,
Valter

**Comment by Cabecinha84** [comment]
Fork updated and conflicts resolved.

**Comment by vikmeup** [comment]
@Cabecinha84 can you rebase and make sure CI looks green?

File: swift/Sources/Addresses/CoinType+Address.swift [hunk 1]
```diff
@@ -96,7 +96,7 @@ public extension CoinType {
             return Set([P2SHPrefix.dash.rawValue, P2PKHPrefix.dash.rawValue])
         case .zcoin:
             return Set([P2SHPrefix.zcoin.rawValue, P2PKHPrefix.zcoin.rawValue])
-        case .zcash:
+        case .zelcash, .zcash:
             return Set([P2SHPrefix.zcashT.rawValue, P2PKHPrefix.zcashT.rawValue])
```
Comment (vikmeup) [inline swift/Sources/Addresses/CoinType+Address.swift]: what's the decision behind making the same prefix as zcash?

Comment (Cabecinha84) [reply swift/Sources/Addresses/CoinType+Address.swift]: Hello vikmeup.
We had no reason to change it as the address prefix makes sense. t for transparent, z for zk-snarks private. Our Acadia "Sapling" private addresses differ from Zcash. But transparent are the same.
Best regards,
Valter

**Comment by Cabecinha84** [comment]
Solved all conflicts. Thank you.

File: README.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ Trust Wallet Core is a cross-platform library that implements low-level cryptogr
 <a href="https://www.dash.org/" target="_blank"><img src="https://raw.githubusercontent.com/TrustWallet/tokens/master/coins/5.png" width="32" /></a>
 <a href="https://zcoin.io/" target="_blank"><img src="https://raw.githubusercontent.com/TrustWallet/tokens/master/coins/136.png" width="32" /></a>
 <a href="https://z.cash/" target="_blank"><img src="https://raw.githubusercontent.com/TrustWallet/tokens/master/coins/133.png" width="32" /></a>
+<a href="https://zel.cash/" target="_blank"><img src="https://raw.githubusercontent.com/TrustWallet/tokens/master/coins/19167.png" width="32" /></a>
```
Comment (hewigovens) [inline README.md]: please move it to end of the list

**Comment by hewigovens** [comment]
please fix CI, first time to see all red :(

**Comment by Cabecinha84** [comment]
Thanks for your feedback. I`ve fixed the tests and change readme to put zel at the end. But I can see in the PR build that wallet core android and wallet core ios is failing. Can you help with those?

**Comment by vikmeup** [comment]
@Cabecinha84 you need to implement tests on iOS and Android.

Swift example where is failed:
```
CompileSwift normal x86_64 /Users/vsts/agent/2.150.3/work/1/s/swift/Tests/CoinAddressDerivationTests.swift
```

**Comment by Cabecinha84** [comment]
Thank you very much for your inputs and help. Despite all checks have passed I want to make a clean PR, so I will close this one and make a new fork and a clean PR after.
Thank you.

**Pull request closed without merging**